File: leetcode-submissions/58_Length_of_Last_Word.cpp
class Solution {
public:
    int lengthOfLastWord(string s) {
        int length=0;
        for(int i=s.size()-1;i>=0;i--){
            if(s[i]!=' '){
                int j=i;
                while(s[j]!=' '){
                    length++;
                    j--;
                }
            break;
            }
        }
        return length;
    }
};

File: leetcode-submissions/58_Length_of_Last_Word.ts
function lengthOfLastWord(s: string): number {
    let j:number =s.length-1;
    
    //this loop removes the spaces and move the pointer to the last index of the last word 
    while(j>=0){
        if(s[j]==" "){
            j--;
        }else{
            break;
        }
    }
    let i:number =j;
    while (i>=0)
    {
        if(s[i]==" "){
            break;
        }
        i--;

    }
    return j-i;
};